1_prep/src/munge_nmls.R

```diff
@@ -46,4 +47,28 @@ munge_nmls <- function(nml_list_rds, lake_ids, base_nml) {
     })
   
   return(nml_objs)
-}
\ No newline at end of file
+}
+
+#' @Title Resample incoming hypsography to 1-m intervals
+#' @decription Resample the H and A values to 1-meter intervals; calc bsn_vals
+#' @param H the vector of H (elevation) values from the lake-specific 
+#' nml parameters loaded in from the nml_list.rds from lake-temperature-model-prep
+#' @param A the vector of A (area) values from the lake-specific nml 
+#' parameters loaded in from the nml_list.rds from lake-temperature-model-prep
+#' @return a list containing H and A vectors and a bsn_vals scaler
+resample_H_A <- function(H, A) {
```

Comment:
Shoot, Sorry @lindsayplatt I thought I had these code changes sequestered in a [different branch](https://github.com/hcorson-dosch/lake-temperature-process-models/tree/resample_hypso), but I guess I messed up the git piece somehow ☹. I had a bunch of uncommitted changes and went through creating new branches and committing to each just the changes that pertained to that branch, but I guess I went awry somewhere/that didn't work?

Reply:
yes... I definitely did that. Shoot. I thought I couldn't go back to main, b/c it would want me to `git stash` anything that was uncommitted first?

Reply:
You can run `git log --graph --decorate --oneline` to visualize your branches. I imagine yours will all be nested?

Reply:
Hard to interpret, but maybe?
```
$ git log --graph --decorate --oneline
* a4cec49 (HEAD -> package_output, origin/package_output) state also included
* a906a5c Fix function documentation
* b159ef4 Remove code to subset to smaller test set of sites
* 5d8ba49 Fix missed step to delete sim_dirs for successful runs
* 953c430 clean up comments
* dd306d4 remove 3_visualize phase for now
* 7b4008c Create 3_extract step and zip up output feathers
* 690cf51 Use site_id in place oflake_id
* 9d4e652 switch to use site_id in place of lake_id
* 2bd1818 Add 4_export step
* 658e786 Use new lake_cell_tile xwalk
* 8ee404f update output processing function documentation
* 5d68f7b (origin/extraction_error_catch, extraction_error_catch) add code to get output extraction errors
* f6960ec (origin/resample_hypso, resample_hypso) add step to resample hypso to 1m intervals
* 8e1e786 (plot_results) rough plotting code, not yet in targets
```

Reply:
Hmmm that output isn't as useful as mine when I run that code.

```
$ git log --graph --decorate --oneline
* 537cf09 (HEAD -> main) switch to using `arrow` not `feather`
*   4571bae (upstream/master) Merge pull request #55 from lindsayplatt/ice_on_fix
|\
| *   74f06fe (origin/ice_on_fix) Merge pull request #8 from lindsayplatt/lindsayplatt-ice_on_fix
| |\
| | *   3f1289e (origin/lindsayplatt-ice_on_fix) Merge branch 'ice_on_fix' of github.com:lindsayplatt/lake-temperature-out into lindsayplatt-ice_on_fix
| | |\
| | |/
| |/|
| * | 054acda (ice_on_fix) return NA for ice_on when there is no data available for previous year
| * | fe17930 return NA for winter_dur_0_4 if there isn't enough data to include the previous year
| | * 85ddfad add multiple tries to the loop for annual metrics
| | * f55ef11 force re-download by changing date
| |/
| * 0609d53 ice_off fix when there is no ice off
| * 359136e hide the grouping messages
| * 7bc752d adjustments so that schmidt_daily_annual_sum can use the next winter's ice_on date
| * 5a6959a ice_onoff_date is a data frame
| * bd09395 update open_water_duration calculation to use the next year's ice_on_date
| * 5997d7a revert & actually combine ice on and ice off calculations into one functions so they can share some of the calculations
| * 648b880 split get_ice_onoff into get_ice_on and get_ice_off
|/
*   484dc91 Merge pull request #54 from lindsayplatt/date_under_5degC
|\
| * a3534cc (origin/date_under_5degC, date_under_5degC) update the comment about returning NAs for calc_first_day_below_temp to be more accurate
```